service-api/src/main/java/greencity/dto/CreateAddressRequestDto.java

```diff
@@ -71,4 +75,31 @@ public class CreateAddressRequestDto {
     @Pattern(regexp = STREET_REGEXP, message = STREET_VALIDATION_MESSAGE)
     @NotEmpty(message = notEmptyValidationMessage)
     private String streetEn;
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+        CreateAddressRequestDto that = (CreateAddressRequestDto) o;
+        return Objects.equals(districtEn.trim().toLowerCase(), that.districtEn.trim().toLowerCase()) &&
+            Objects.equals(district.trim().toLowerCase(), that.district.trim().toLowerCase()) &&
+            Objects.equals(regionEn.trim().toLowerCase(), that.regionEn.trim().toLowerCase()) &&
+            Objects.equals(region.trim().toLowerCase(), that.region.trim().toLowerCase()) &&
+            Objects.equals(houseNumber, that.houseNumber) &&
+            Objects.equals(entranceNumber, that.entranceNumber) &&
+            Objects.equals(houseCorpus, that.houseCorpus) &&
+            Objects.equals(coordinates, that.coordinates) &&
+            Objects.equals(city.trim().toLowerCase(), that.city.trim().toLowerCase()) &&
+            Objects.equals(cityEn.trim().toLowerCase(), that.cityEn.trim().toLowerCase()) &&
+            Objects.equals(street.trim().toLowerCase(), that.street.trim().toLowerCase()) &&
+            Objects.equals(streetEn.trim().toLowerCase(), that.streetEn.trim().toLowerCase());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(districtEn, district, regionEn, region, houseNumber, entranceNumber,
```

Comment:
should we use `trim` here too

Reply:
it brakes `hashCode()` and `hashCode()` contract
`Equals Consistency: If two objects are equal according to the equals() method, then the hashCode() method must return the same value for them.`